tests/integration/test_logcollector/test_options/test_restrict_regex.py

```diff
@@ -0,0 +1,269 @@
+    - https://documentation.wazuh.com/current/user-manual/reference/internal-options.html#logcollector
+
+tags:
+    - logcollector_options
+'''
+import os
+import sys
+import re
+import pytest
+
+from wazuh_testing.tools import PREFIX
+from wazuh_testing.tools.local_actions import run_local_command_returning_output
+from wazuh_testing.tools.configuration import load_configuration_template, get_test_cases_data
+from wazuh_testing.modules.logcollector import event_monitor as evm
+from wazuh_testing.modules import logcollector as lc
+
+
+# Reference paths
+TEST_DATA_PATH = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+CONFIGURATIONS_PATH = os.path.join(TEST_DATA_PATH, 'configuration_template')
+TEST_CASES_PATH = os.path.join(TEST_DATA_PATH, 'test_cases')
+
+# Configuration and cases data
+t1_configurations_path = os.path.join(CONFIGURATIONS_PATH, 'configuration_restrict_regex_default.yaml')
+t1_cases_path = os.path.join(TEST_CASES_PATH, 'cases_restrict_regex_default.yaml')
+
+t2_configurations_path = os.path.join(CONFIGURATIONS_PATH, 'configuration_restrict_regex_type_values.yaml')
+t2_cases_path = os.path.join(TEST_CASES_PATH, 'cases_restrict_regex_type_values.yaml')
+
+# Test configurations
+test_file = os.path.join(PREFIX, 'test')
+
+t1_configuration_parameters, t1_configuration_metadata, t1_case_ids = get_test_cases_data(t1_cases_path)
+for count, value in enumerate(t1_configuration_parameters):
+    t1_configuration_parameters[count]['LOCATION'] = test_file
+t1_configurations = load_configuration_template(t1_configurations_path, t1_configuration_parameters,
+                                                t1_configuration_metadata)
+
+t2_configuration_parameters, t2_configuration_metadata, t2_case_ids = get_test_cases_data(t2_cases_path)
+for count, value in enumerate(t2_configuration_parameters):
+    t2_configuration_parameters[count]['LOCATION'] = test_file
+t2_configurations = load_configuration_template(t2_configurations_path, t2_configuration_parameters,
+                                                t2_configuration_metadata)
+
+prefix = lc.LOG_COLLECTOR_PREFIX
+
+
+# Tests
+@pytest.mark.tier(level=1)
+@pytest.mark.parametrize('new_file_path,', [test_file], ids=[''])
+@pytest.mark.parametrize('local_internal_options,', [lc.LOGCOLLECTOR_DEFAULT_LOCAL_INTERNAL_OPTIONS], ids=[''])
+@pytest.mark.parametrize('configuration, metadata', zip(t1_configurations, t1_configuration_metadata), ids=t1_case_ids)
+def test_restrict_default(configuration, metadata, new_file_path, create_file, truncate_monitored_files,
+                          local_internal_options, set_wazuh_configuration_with_local_internal_options,
+                          restart_wazuh_function):
+    '''
+    description: Check if logcollector reads or ignores a log according to a regex configured in the restrict tag for a
+    given log file.
```

Comment:
Please keep the identation.

Reply:
fixed in https://github.com/wazuh/wazuh-qa/pull/3582/commits/736cebbf8d76781caa2709ed9b8f305e31183c79